prism/src/main/java/org/apache/falcon/service/FeedSLAAlertService.java

```diff
@@ -117,33 +114,37 @@ void processSLACandidates(){
         try{
             for (PendingInstanceBean pendingInstanceBean : pendingInstanceBeanList) {
 
-                String feedName = pendingInstanceBean.getFeedName();
+                String entityName = pendingInstanceBean.getEntityName();
                 String clusterName = pendingInstanceBean.getClusterName();
                 Date nominalTime = pendingInstanceBean.getNominalTime();
-                Feed feed = ConfigurationStore.get().get(EntityType.FEED, feedName);
+                String entityType = pendingInstanceBean.getEntityType();
 
-                Cluster cluster =  FeedHelper.getCluster(feed, clusterName);
+                org.apache.falcon.entity.v0.cluster.Cluster cluster = ClusterHelper.getCluster(clusterName);
 
                 Set<SchedulableEntityInstance> schedulableEntityInstances= FeedSLAMonitoringService.get().
-                        getFeedSLAMissPendingAlerts(feed.getName(), cluster.getName(), nominalTime, nominalTime);
+                        getFeedSLAMissPendingAlerts(entityName, cluster.getName(), nominalTime, nominalTime
+                                , entityType);
                 if (schedulableEntityInstances.isEmpty()){
-                    store.deleteFeedAlertInstance(feed.getName(), cluster.getName(), nominalTime);
+                    store.deleteFeedAlertInstance(entityName, cluster.getName(), nominalTime,
+                            entityType);
                     return;
                 }
                 List<SchedulableEntityInstance> schedulableEntityList = new ArrayList<>(schedulableEntityInstances);
                 SchedulableEntityInstance schedulableEntityInstance = schedulableEntityList.get(0);
 
 
                 if (schedulableEntityInstance.getTags().contains(FeedSLAMonitoringService.get().TAG_WARN)) {
-                    store.putSLAAlertInstance(feedName, clusterName, nominalTime, true, false);
+                    store.putSLAAlertInstance(entityName, clusterName, EntityType.FEED.toString(),
```

Comment:
you can use entityType variable rather than hardcode FEED. I understand that this branch is possible for feed.